lib/krane/kubernetes_resource/stateful_set.rb

```diff
@@ -19,25 +22,34 @@ def status
     end
 
     def deploy_succeeded?
-      success = observed_generation == current_generation &&
-        desired_replicas == status_data['readyReplicas'].to_i &&
-        status_data['currentRevision'] == status_data['updateRevision']
+      success = observed_generation == current_generation
+
+      @pods.each do |pod|
+        success &= pod.definition["metadata"]["labels"]["controller-revision-hash"] == status_data['updateRevision']
+      end
+
       if update_strategy == ONDELETE
-        # Gem cannot monitor update since it doesn't occur until delete
         unless @success_assumption_warning_shown
           @logger.warn("WARNING: Your StatefulSet's updateStrategy is set to OnDelete, "\
-                       "which means updates will not be applied until its pods are deleted. "\
-                       "Consider switching to rollingUpdate.")
+                       "which means updates will not be applied until its pods are deleted.")
           @success_assumption_warning_shown = true
         end
+
+        if required_rollout == 'full'
```

Comment:
can we use the constant you defined at the top of this file?
```suggestion
        if required_rollout == DEFAULT_REQUIRED_ROLLOUT
```